haystack/preview/components/generators/openai/chatgpt.py

```diff
@@ -0,0 +1,276 @@
+from typing import Optional, List, Callable, Dict, Any
+
+import sys
+import builtins
+import logging
+from dataclasses import asdict
+
+import openai
+
+from haystack.preview import component, default_from_dict, default_to_dict, DeserializationError
+
+# from haystack.preview.llm_backends.openai.chatgpt import ChatGPTBackend
+from haystack.preview.dataclasses.chat_message import ChatMessage
+
+
+logger = logging.getLogger(__name__)
+
+
+TOKENS_PER_MESSAGE_OVERHEAD = 4
+
+
+def default_streaming_callback(chunk):
+    """
+    Default callback function for streaming responses from OpenAI API.
+    Prints the tokens of the first completion to stdout as soon as they are received and returns the chunk unchanged.
+    """
+    if hasattr(chunk.choices[0].delta, "content"):
+        print(chunk.choices[0].delta.content, flush=True, end="")
+    return chunk
```

Comment:
This seems unused.
```suggestion
```

Reply:
It's used in the tests, and I'd leave it as an example for other callbacks, to use for example in demos. Should I still remove it?

Reply:
If it's used in tests it should be moved there I think. Leaving it here makes it part of the public API.

Reply:
We can add it later if necessary, I fear we'd have a proliferation of default streaming handlers all over the place otherwise.